File: solutions/cpp/etl/1/etl.cpp
#include "etl.h"

namespace etl {
    auto transform(const std::map<int, std::vector<char>>& old) -> std::map<char, int> {
        return std::accumulate(
                    old.begin(),
                    old.end(),
                    std::map<char, int>{},
                    [](auto&& acc, const auto& pair) {
                        int value = pair.first;

                        for (const auto& letter : pair.second) {
                            acc.try_emplace(std::tolower(letter), value);
                        }

                        return acc;
                    }
                );
    }
}  // namespace etl

File: solutions/javascript/etl/1/etl.js
export const transform = (old) => {
  return Object.entries(old)
    .reduce((acc, [val, keys]) => {
      return keys.reduce((acc2, key) => { 
        return { ...acc2, [key.toLowerCase()] : parseInt(val) };
      }, acc);
    }, {});
};
